src/plugins/intel_cpu/src/nodes/sparse_fill_empty_rows.cpp

```diff
@@ -0,0 +1,153 @@
+    if (!supportedPrimitiveDescriptors.empty()) {
+        return;
+    }
+    ov::element::Type valuesPrecision = getOriginalInputPrecisionAtPort(0);
+    ov::element::Type indicesPrecision = getOriginalInputPrecisionAtPort(2);
+    addSupportedPrimDesc({{LayoutType::ncsp, valuesPrecision},        // values
+                          {LayoutType::ncsp, indicesPrecision},       // dense_shape
+                          {LayoutType::ncsp, indicesPrecision},       // indices
+                          {LayoutType::ncsp, valuesPrecision}},       // default_value
+                         {{LayoutType::ncsp, indicesPrecision},       // output_indices
+                          {LayoutType::ncsp, valuesPrecision},        // output_values
+                          {LayoutType::ncsp, ov::element::boolean}},  // empty_row_indicator
+                         impl_desc_type::ref);
+}
+
+bool SparseFillEmptyRows::created() const {
+    return getType() == Type::SparseFillEmptyRows;
+}
+
+bool SparseFillEmptyRows::needPrepareParams() const {
+    return false;
+}
+
+bool SparseFillEmptyRows::isExecutable() const {
+    return !isInputTensorAtPortEmpty(1) && !isInputTensorAtPortEmpty(3);
+}
+
+void SparseFillEmptyRows::executeDynamicImpl(const dnnl::stream& strm) {
+    const auto& valuesMemory = getSrcMemoryAtPort(0);
+    const auto& indicesMemory = getSrcMemoryAtPort(2);
+    const auto& valuesShape = valuesMemory->getShape();
+    const auto& indicesShape = indicesMemory->getShape();
+
+    const auto indicesPrecision = getParentEdgeAt(2)->getMemory().getDesc().getPrecision();
+    int64_t numRows = 0;
+
+    if (indicesPrecision == ov::element::i32) {
+        const auto* denseShapePtr = getSrcDataAtPortAs<const int32_t>(1);
+        numRows = static_cast<int64_t>(denseShapePtr[0]);
+    } else {
+        const auto* denseShapePtr = getSrcDataAtPortAs<const int64_t>(1);
+        numRows = denseShapePtr[0];
+    }
+
+    std::unordered_set<int64_t> existingRows;
+    size_t indicesCount = indicesShape.getElementsCount() / 2;  // Divide by 2 because indices is [M, 2]
+
+    if (indicesPrecision == ov::element::i32) {
+        const auto* indicesPtr = getSrcDataAtPortAs<const int32_t>(2);
+        for (size_t i = 0; i < indicesCount; i++) {
+            existingRows.insert(static_cast<int64_t>(indicesPtr[i * 2]));
+        }
+    } else {
+        const auto* indicesPtr = getSrcDataAtPortAs<const int64_t>(2);
+        for (size_t i = 0; i < indicesCount; i++) {
+            existingRows.insert(indicesPtr[i * 2]);
+        }
+    }
+
+    size_t emptyRowsCount = numRows - existingRows.size();
+    size_t valuesCount = valuesShape.getElementsCount();
+    ov::Shape outputIndicesShape{valuesCount + emptyRowsCount, 2};
+    ov::Shape outputValuesShape{valuesCount + emptyRowsCount};
+    ov::Shape emptyRowIndicatorShape{static_cast<size_t>(numRows)};
+
+    redefineOutputMemory({outputIndicesShape, outputValuesShape, emptyRowIndicatorShape});
```

Comment:
This part looks like duplication of shape inference, would it be possible to replace that with proper implementation of "needShapeInfer" in the CPU SparseFillEmptyRows?

I would recommend to verify this approach, but to me it's not a blocker.